src/NumericalAlgorithms/Interpolation/Callbacks/ObserveSurfaceIntegrals.hpp

```diff
@@ -69,32 +122,36 @@ struct ObserveSurfaceIntegrals {
         get<StrahlkorperTags::NormalOneForm<Frame>>(box),
         get<StrahlkorperTags::Radius<Frame>>(box),
         get<StrahlkorperTags::Rhat<Frame>>(box));
-    std::vector<double> time_and_integrals(tmpl::size<TagsToObserve>::value +
-                                           1);
+
+    // The +1 in the sizes below is because `TagsToObserve` contains
+    // only the integrals, but both `legend` and `time_and_integrals` contain
+    // time in addition to the integrals.
     std::vector<std::string> legend(tmpl::size<TagsToObserve>::value + 1);
+    std::array<double, tmpl::size<TagsToObserve>::value + 1>
+        time_and_integrals{};
     time_and_integrals[0] = temporal_id.time().value();
     legend[0] = "Time";
     size_t s = 1;
-    tmpl::for_each<TagsToObserve>([&](auto tag) noexcept {
-      using Tag = typename decltype(tag)::type;
+    tmpl::for_each<TagsToObserve>([&](auto tmpl_for_tag) noexcept {
```

Comment:
maybe `tmpl_for_tag` -> `tag_v` since that is consistent-ish with the STL's type traits. I find myself not understanding what `tmpl_for_tag` actually means since I'm reading it as `template metaprogramming library for tag`